mvc/views/tag-helpers/authoring.rst

```diff
@@ -1,8 +1,384 @@
+   :language: aspx-cs
+   :emphasize-lines: 15-16
+   :lines: 1-17
+   
+4. Run the app and use your favorite browser to view the HTML source so you can verify that the email tag is replaced with anchor markup (``<a>Support</a>``). Support is rendered as a link, but it doesn't have an ``href`` attribute to make it functional. We'll fix that in the next section.
+
+Note: Like `HTML tags and attributes <http://www.w3.org/TR/html-markup/documents.html#case-insensitivity>`__, tags, class names and attributes in Razor, and C# are not case-sensitive.
+
+A working email Tag Helper
+----------------------------------
+In this section, we will update the ``EmailTagHelper`` so that it will create a valid anchor tag for email. We'll update our tag helper to take information from a Razor view (in the form of a ``mail-to`` attribute) and use that in generating the anchor.
+
+Update the ``EmailTagHelper`` class with the following:
+
+.. literalinclude:: authoring/sample/TagHlp/src/AuthoringTagHelpers/TagHelpers/EmailTagHelperMailTo.cs
+   :lines: 5-24
+   :dedent: 3 
+
+Notes:
+
+- The ASP.NET 5 runtime translates C# Pascal-cased classes and property names for tag helpers into `lower kebab case <http://stackoverflow.com/questions/11273282/whats-the-name-for-dash-separated-case/12273101#12273101>`__. Therefore, to use the ``MailTo`` attribute, you'll use ``<email mail-to="value"/>``.
+- The following line shows how to set attributes:
+
+.. literalinclude:: authoring/sample/TagHlp/src/AuthoringTagHelpers/TagHelpers/EmailTagHelperMailTo.cs
+   :lines: 7-24
+   :dedent: 3
+   :emphasize-lines: 15 
+
+3. Update the markup in the *Views/Home/Contact.cshtml* file with these changes:
+
+.. literalinclude::  authoring/sample/TagHlp/src/AuthoringTagHelpers/Views/Home/ContactCopy.cshtml
+   :language: aspx-cs
+   :emphasize-lines: 15-16 
+   :lines: 1-17
+ 
+4. Run the app and verify that it generates the correct links.
+
+**Note:** If you were to write the email tag self-closing (``<email mail-to="Rick" />``), the final output would also be self-closing. To write this as void structured tag helper you must decorate the class with the following attribute:
+
+.. literalinclude:: authoring/sample/TagHlp/src/AuthoringTagHelpers/TagHelpers/EmailTagHelperMailVoid.cs
+   :lines: 7
+
+In our example, the output would be ``<a href="mailto:Rick@contoso.com" />``. Self-closing anchor tags (also known as `void elements <http://www.w3.org/TR/html5/syntax.html#void-elements>`__) are not valid HTML, so you wouldn't want to create one, but you might want to create a tag helper for a void element. The ASP.NET 5 runtime sets the type of the ``TagMode`` property after reading a tag.
+
+In this section we will update the ``EmailTagHelper`` so that it gets the target ``mail-to`` from the content. 
+
+#. Update the ``EmailTagHelper`` class with the following code:
+
+.. literalinclude:: authoring/sample/TagHlp/src/AuthoringTagHelpers/TagHelpers/EmailTagHelper.cs
+   :lines: 6-18
```

Comment:
This is the 3rd update of the class, I didn't include the entire file for the 2nd because I had to change the namepace (because I have 4 versions of this class). I can't do it here for the same reason, I don't want to show a different namespace. The using haven't changed from the first occurrence of the class.